openquake/hazardlib/gsim/nshmp_2014.py

```diff
@@ -0,0 +1,269 @@
+# The Hazard Library
+# Copyright (C) 2015 GEM Foundation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Module exports :class:`AbrahamsonEtAl2014NSHMPUpper`
+               :class:`AbrahamsonEtAl2014NSHMPLower`
+               :class:`BooreEtAl2014NSHMPUpper`
+               :class:`BooreEtAl2014NSHMPLower`
+               :class:`CampbellBozorgnia2014NSHMPUpper`
+               :class:`CampbellBozorgnia2014NSHMPLower`
+               :class:`ChiouYoungs2014NSHMPUpper`
+               :class:`ChiouYoungs2014NSHMPLower`
+               :class:`Idriss2014NSHMPUpper`
+               :class:`Idriss2014NSHMPLower`
+"""
+from __future__ import division
+import numpy as np
+# standard acceleration of gravity in m/s**2
+from scipy.constants import g
+
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, SA
+# NGA West 2 GMPEs
+from openquake.hazardlib.gsim.abrahamson_2014 import AbrahamsonEtAl2014
+from openquake.hazardlib.gsim.boore_2014 import BooreEtAl2014
+from openquake.hazardlib.gsim.campbell_bozorgnia_2014 import \
+    CampbellBozorgnia2014
+from openquake.hazardlib.gsim.chiou_youngs_2014 import ChiouYoungs2014
+from openquake.hazardlib.gsim.idriss_2014 import Idriss2014
+
+
+def nga_west2_epistemic_adjustment(magnitude, distance):
+    """
+    Applies the "average" adjustment factor for epistemic uncertainty
+    as defined in Table 17 of Petersen et al., (2014)
+
+                   R < 10.  | 10.0 <= R < 30.0  |    R >= 30.0
+    -----------------------------------------------------------
+      M < 6.0   |   0.37    |      0.22         |       0.22
+    6 <= M <7.0 |   0.25    |      0.23         |       0.23
+      M >= 7.0  |   0.40    |      0.36         |       0.33
+    """
+    if magnitude < 6.0:
```

Comment:
THE GMPE tables were created by parsing the existing GMPE tables for the non-adjusted GMPEs, applying the above factors and then writing to new tables. These don't guarantee correctness, but should ensure coverage.